test/jMeter/generateTestJMeter.py

```diff
@@ -0,0 +1,155 @@
+
+BODY_DATA = """
+{
+    "brandName": {
+        "type": "Property",
+        "value": "Chevrolet Spin"
+    }
+}
+"""
+# defineParams: Create a structure of folders for HTML Report or just .csv file
+# @params : isCreateHTMLReport - Works like a switch for choose the structure
+
+
+def defineParams(tree):
+    root = tree.getroot()
+
+    verbSelected = list(filter(lambda verb: verb == HTTP_VERB, HTTP_VERBS))
+
+    if verbSelected == []:
+        print('ERROR:', 'Use a Orion-LD Broker supported HTTP verb.')
+        os.abort()
+
+    for item in root.iter('stringProp'):
+        if item.attrib['name'] == "ThreadGroup.num_threads":
+            item.text = str(NUM_OF_THREADS)
+        if item.attrib['name'] == "LoopController.loops":
+            item.text = str(LOOPS)
+        if item.attrib['name'] == "ThreadGroup.ramp_time":
+            item.text = str(RAMP_TIME)
+        if item.attrib['name'] == "HTTPSampler.domain":
+            item.text = str(DOMAIN_IP)
+        if item.attrib['name'] == "HTTPSampler.port":
+            item.text = str(PORT)
+        if item.attrib['name'] == "HTTPSampler.path":
+            item.text = str(PATH_SERVICE)
+        if item.attrib['name'] == "HTTPSampler.method":
+            item.text = str(HTTP_VERB)
+
+        if HTTP_VERB == "GET":
+            if item.attrib['name'] == "Argument.value":
+                item.text = ''
+        elif HTTP_VERB == "POST":
+            if item.attrib['name'] == "Argument.value":
+                item.text = BODY_DATA
+        elif HTTP_VERB == "PATCH" or HTTP_VERB == "DELETE":
+            if item.attrib['name'] == "Header.value":
+                item.text = 'application/json'
+            if item.attrib['name'] == "Argument.value":
+                item.text = BODY_DATA
+
+
+# createFolders: Create a structure of folders for HTML Report or just .csv file
+# @params : isCreateHTMLReport - Works like a switch for choose the structure
+
+
+def createFolders(isCreateHTMLReport):
+    if not os.path.exists('output'):
+        os.mkdir('output')
+    if isCreateHTMLReport:
+        os.mkdir('output/' + OUTPUT_FILE_NAME + '_Report')
+        os.mkdir('output/' + OUTPUT_FILE_NAME + '_Report/Report')
+    else:
+        os.mkdir('output/' + OUTPUT_FILE_NAME)
+
+# executeTest: execute one of the options of menu for JMeter Test
+
+
+def executeTest(operation, tree):
+    if operation == GENERATE_COMPLETE_REPORT:
+        createFolders(True)
+        tree.write('output/' + OUTPUT_FILE_NAME +
+                   '_Report/' + OUTPUT_FILE_NAME + '.jmx')
+        os.system('jmeter -n -t output/' + OUTPUT_FILE_NAME + '_Report/' + OUTPUT_FILE_NAME + '.jmx -l output/' + OUTPUT_FILE_NAME +
+                  '_Report/' + OUTPUT_FILE_NAME + '.csv -e -o output/' + OUTPUT_FILE_NAME + '_Report/Report')
+    elif operation == GENERATE_ONLY_CSV_FILE:
+        createFolders(False)
+        tree.write('output/' + OUTPUT_FILE_NAME +
```

Comment:
also you have the `join` method of the os.path and you should not hardcode the `/` ... as path separator

Reply:
Nice tip. I will implement it too

Reply:
NTC